third-culture-software/bhima
# Fix editing shipments (updating packing list)

This PR fixes an issue with editing a shipment (eg, "update packing list" in the action menu of the Shipments registry") that was losing the destination depot in the editing process.

This was a difficult one to track down due to the interaction of the callback functions that were being invoked when the shipment info was loaded.   A part of the fix was to prevent the display of the editing page until the startup was completed (eg, when the shipment info had been fully installed).

Closes https://github.com/IMA-WorldHealth/bhima/issues/6559

**TESTING**
- Use bhima_test
- Create a shipment.  Do NOT mark it ready for shipping.
- In the shipments registry, use the action menu item "Updated Packing List" to edit the shipment.
- Verify that the Destination depot shown and is correct
- Edit the shipment and save it.
- Try editing it again to double-check.
- Create another shipment but do not mark it ready for shipping
- Log out of Bhima, clear the cache, and log back in
- Go to Shipments > New Shipment and notice the popup to select a origin Depot 
   - DO NOT Select a depot. Just cancel.
- Go to the Shipments > Shipments Registry page
- Use the action menu to edit the last shipment created.  Note that there is no prompt to select a depot, since that is determined from the shipment info


**Notes**
- changing the origin depot deletes all items since these may not be available in a different depot.  If needed, they should be re-added.  NOTE: Will add an issue to prevent changing the origin depot.
- Changing the destination depot does not delete the packing list items
